champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/mage/axe/Fissure.java

```diff
@@ -0,0 +1,382 @@
+                if (!isForbiddenBlockType(materialBelow) && !UtilBlock.airFoliage(blockBelow)) {
+                    return materialBelow;
+                } else {
+                    return baseBlock.getType();
+                }
+            }
+
+            private void checkForEntityCollisions(Block blockToPlace, int currentIndex, Player player) {
+                List<KeyValue<LivingEntity, EntityProperty>> entities = UtilEntity.getNearbyEntities(player, blockToPlace.getLocation(), 1.0, EntityProperty.ENEMY);
+                for (KeyValue<LivingEntity, EntityProperty> keyValue : entities) {
+                    LivingEntity entity = keyValue.getKey();
+                    onEntityCollision(entity, currentIndex, player);
+                }
+            }
+        }.runTaskTimer(champions, 0L, 1L);
+    }
+
+
+
+    private void onEntityCollision(Entity entity, int blocksTraveled, Player player) {
+
+        if (!(entity instanceof LivingEntity) || entity.equals(player)) {
+            return;
+        }
+
+        LivingEntity ent = (LivingEntity) entity;
+
+        playerHits.putIfAbsent(player, new HashSet<>());
+        HashSet<Player> playersHit = playerHits.get(player);
+
+        if (ent instanceof Player && !playersHit.contains(ent)) {
+
+            int level = getLevel(player);
+            double damage = getDamage(blocksTraveled, level);
+            ent.damage(damage);
+
+            playersHit.add((Player) ent);
+
+            UtilMessage.simpleMessage(player, getClassType().getName(), "You hit <alt2>" + ent.getName() + "</alt2> with <alt>" + getName());
+            UtilMessage.simpleMessage(ent, getClassType().getName(), "<alt2>" + player.getName() + "</alt2> hit you with <alt>" + getName());
+        }
+    }
+
+
+    public void removeFissure(Player player) {
+        HashSet<Block> createdBlocks = playerCreatedBlocks.get(player);
+        if (createdBlocks != null) {
+            for (Block block : createdBlocks) {
+                Material blockMaterial = block.getType();
+                if(block.getType() != Material.AIR){
+                block.setType(Material.AIR);
+                player.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, blockMaterial);
+                }
+            }
+            createdBlocks.clear();
+        }
+
+        playerFissurePaths.remove(player);
+        playerHits.remove(player);
+        playerCreatedBlocks.remove(player);
+    }
+
+    private boolean isForbiddenBlockType(Material material) {
+        Set<Material> forbiddenBlockTypes = EnumSet.of(
+                Material.TNT,
+                Material.ANVIL,
+                Material.ENCHANTING_TABLE
+        );
+
+        if (forbiddenBlockTypes.contains(material)) {
+            return true;
+        }
+
+        String materialName = material.name().toUpperCase();
+
+        if (materialName.contains("DOOR") || materialName.contains("GATE")) {
+            return true;
+        }
+
+        return false;
```

Comment:
block.getState() instanceof Container

Feel like this should be a List loaded from config.